src/test/java/seedu/guest/model/guest/BillTest.java

```diff
@@ -0,0 +1,59 @@
+package seedu.guest.model.guest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.guest.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class BillTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Bill(null));
+    }
+
+    @Test
+    public void constructor_invalidBill_throwsIllegalArgumentException() {
+        String invalidBill = "";
+        assertThrows(IllegalArgumentException.class, () -> new Bill(invalidBill));
+    }
+
+    @Test
+    public void isValidBill() {
+        // null bill
+        assertThrows(NullPointerException.class, () -> Bill.isValidBill(null));
+
+        // invalid bill
+        assertFalse(Bill.isValidBill("")); // empty string
+        assertFalse(Bill.isValidBill(" ")); // spaces only
+        assertFalse(Bill.isValidBill("abc")); // non-numerical input
+        assertFalse(Bill.isValidBill("-1")); // negative number
+        assertFalse(Bill.isValidBill(".99")); // start with decimal point
+        assertFalse(Bill.isValidBill("1.")); // end in decimal point
+        assertFalse(Bill.isValidBill("1.5")); // only 1 decimal place
+        assertFalse(Bill.isValidBill("1.500")); // more than 2 decimal places
+        assertFalse(Bill.isValidBill("1,000")); // comma
```

Comment:
Should there be a case of where the bill is "1 000", ie a space as part of the string?

Reply:
Good suggestion, added!